Typescript error: Cannot find name 'Buffer'/'http'/'https'

Hi!

First, I want to thank you for this SDK, it will be very useful for us!

I have some errors when building the javascript bundle for my application (React/Typescript/Webpack). It seems to work anyway (or maybe I have not used it enough yet), but I am not comfortable building our app with errors :).

Here is an extract of the errors:

```
Hash: 6824b5df64cd8f5d85c3
Version: webpack 1.14.0
Time: 4139ms
          Asset     Size  Chunks             Chunk Names
    frontend.js  3.63 MB       0  [emitted]  main
frontend.js.map  5.28 MB       0  [emitted]  main
    + 617 hidden modules

ERROR in /Users/rmarenco/Dev/taiga/frontend/node_modules/aws-sdk/clients/wafregional.d.ts
(416,39): error TS2304: Cannot find name 'Buffer'.

...

ERROR in /Users/rmarenco/Dev/taiga/frontend/node_modules/aws-sdk/lib/config.d.ts
(1,34): error TS2307: Cannot find module 'http'.

ERROR in /Users/rmarenco/Dev/taiga/frontend/node_modules/aws-sdk/lib/config.d.ts
(2,35): error TS2307: Cannot find module 'https'.
```

And you can find the full errors on this gist => https://gist.github.com/remimarenco/c51c3170fbfd8d548e6fd2fe4542cdb1

I am also including my configuration files in the gist, in case they could be useful :). The architecture of my application is like this:

taiga/
  - frontend/
    - node_modules/ <= All my frontend dependencies
    - src/ <= All my .tsx
    - `package.json`
  - node_modules/ <= All my backend and dev dependencies
  - taiga2/
    - static/ <= Folder where webpack store the outputs
  - `package.json`
  - `tsconfig.json`
  - `webpack.config.js`

Do you think you know what is not correct in my configuration? Or is it on the sdk side + webpack (and/or typescript)?

Thanks a lot!

--- a/README.md
+++ b/README.md
@@ -124,6 +124,14 @@ var AWS = require('aws-sdk/global');
 var S3 = require('aws-sdk/clients/s3');
 ```
 
+### With Angular
+Due to the SDK's reliance on node.js typings, you may encounter compilation 
+[issues](https://github.com/aws/aws-sdk-js/issues/1271) when using the
+typings provided by the SDK in an Angular project created using the Angular CLI.
+
+To resolve these issues, either add `"types": ["node"]` to the project's `tsconfig.app.json`
+file, or remove the `"types"` field entirely.
+
 ### Known Limitations
 There are a few known limitations with the bundled TypeScript definitions at this time:
 

